### PR TITLE
Fix exposing options of MeshDebugPluginMaterial

### PR DESCRIPTION
**Pull request by alexchuber**
I had believed PR #14127 would allow for a user to update `options` after initialization, but I forgot that the `expandToProperty` decorator won't work as expected for custom objects like `options`.  

Here's a fix. The plugin no longer stores the options object, and instead unwraps each property of option into its own direct plugin property.

It also means I can remove the serialize/parse methods, applying the serialization decorator to each property instead.

(I did look into making a new decorator for custom objects, but I don't know enough about TS metaprogramming to make it in time!)

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14139/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14139/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14139/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14139/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14139/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14139/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14139/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14139/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14139/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Materials/meshDebugPluginMaterial.ts [hunk 1]
```diff
@@ -332,13 +332,122 @@ export class MeshDebugPluginMaterial extends MaterialPluginBase {
...
+    private _multiply: boolean;
+    /**
+     * Whether the mesh debug visualization should multiply with color underneath.
+     * Defaults to true.
+     */
+    @serialize()
+    @expandToProperty("_markAllDefinesAsDirty")
+    public multiply: boolean;
+
+    private _shadedDiffuseColor: Color3;
```
Comment (Popov72) [inline packages/dev/core/src/Materials/meshDebugPluginMaterial.ts:353]: In fact, you don't need to rebuild the shader when this property changes, so you don't need the `@expandToProperty("_markAllDefinesAsDirty")` construction.

You can simply replace:
```typescript
    private _shadedDiffuseColor: Color3;
    @serializeAsColor3()
    @expandToProperty("_markAllDefinesAsDirty")
    public shadedDiffuseColor: Color3;
```
with:
```typescript
    @serializeAsColor3()
    public shadedDiffuseColor: Color3;
```
Same for most properties.

I think you only need the `mode` and `multiply` update to actually recompile the shader.

**Review by Popov72 — CHANGES_REQUESTED** [review]

File: packages/dev/core/src/Materials/meshDebugPluginMaterial.ts (hunk 1 above)
Comment (alexchuber) [reply packages/dev/core/src/Materials/meshDebugPluginMaterial.ts]: Fixed :)

**Review by Popov72 — APPROVED** [review]

**Pull request merged**